libraries/helpers.rb

```diff
@@ -570,6 +570,19 @@ def is_compute_node_bootstrap_timeout_updated?(previous_config, config)
   evaluate_compute_bootstrap_timeout(previous_config) != evaluate_compute_bootstrap_timeout(config)
 end
 
+def evaluate_realmemory_to_ec2memory_ratio(config)
+  require 'json'
+  extra_chef_attrs = config.dig("DevSettings", "Cookbook", "ExtraChefAttributes")
+  JSON.parse(extra_chef_attrs).dig("cluster", "realmemory_to_ec2memory_ratio")
+end
+
+def is_realmemory_to_ec2memory_ratio_updated?
+  require 'yaml'
+  config = YAML.safe_load(File.read(node['cluster']['cluster_config_path']))
+  previous_config = YAML.safe_load(File.read(node['cluster']['previous_cluster_config_path']))
```

Comment:
if you want to keep the check for the update then we should avoid reading the configs in too many places and centralize this logic. fetch_config resource already takes care of loading the current cluster config in the `node['cluster']['config']`. We could extend that logic to also load the previous config and then everywhere we need these objects we access them with `node['cluster']['config']` and `node['cluster']['previous_config']`